File: BackEnd/10_week_Algorithms/1.선형_알고리즘_기초/1C/1C.js
// 배열의 최대값

const readline = require('readline');

// readline 인터페이스 설정
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// 숫자들 저장을 위한 배열
let numbers = [];

// 첫 번째 줄: 숫자의 개수 N
rl.on('line', (line) => {
    if (numbers.length === 0) {
        // 숫자의 개수 N을 읽고, 그 줄은 무시
        const N = parseInt(line);
    } else {
        // 두 번째 줄: 숫자들
        numbers = line.split(' ').map(Number);
        // 가장 큰 숫자를 찾기
        const maxNumber = Math.max(...numbers);
        // 결과 출력
        console.log(maxNumber);
        // 인터페이스 종료
        rl.close();
    }
});

File: Algorithm/10주_완성_알고리즘_코딩테스트_기초/1.선형_알고리즘_기초/1C_배열의_최대값/1C.py
# 배열의 최대값

import sys
input = sys.stdin.read

def find_max_number(data):
    
    # 첫 줄은 숫자의 개수 N (사용하지 않음)
    N = int(data[0])
    
    # 두 번째 줄의 숫자들을 공백으로 구분하여 리스트로 변환
    numbers = list(map(int, data[1:]))
    
    max_number = max(numbers)
    
    print(max_number)


# 입력 읽기
data = input().strip().split()

find_max_number(data)
